python_modules/dagster-graphql/dagster_graphql/schema/instigation.py

```diff
@@ -139,11 +140,34 @@ class Meta:
         name = "DynamicPartitionsRequestType"
 
 
-class GrapheneDynamicPartitionsRequest(graphene.ObjectType):
+class DynamicPartitionsRequestMixin:
     partitionKeys = graphene.List(graphene.NonNull(graphene.String))
     partitionsDefName = graphene.NonNull(graphene.String)
     type = graphene.NonNull(GrapheneDynamicPartitionsRequestType)
 
+    class Meta:
+        name = "DynamicPartitionRequestMixin"
+
+    def get_dynamic_partitions_request(
```

Comment:
I'm not super familiar with this mixin pattern, but this strikes me as a little weird. Why would a request be expected to have a request inside it?

Reply:
A mixin is like a superclass, and subclasses inherit the methods and resolvers of the mixin. I was mostly following the pattern here: https://github.com/dagster-io/dagster/blob/master/python_modules/dagster-graphql/dagster_graphql/schema/pipelines/pipeline.py#L578-L584

The alternative would be to pass in the request object when the mixin is initialized instead.

Reply:
Got it. I think it's worth adding a similar comment here.